ctapipe/image/tests/test_cleaning.py

```diff
@@ -142,3 +142,17 @@ def test_tailcuts_clean_with_isolated_pixels():
                                          boundary_thresh=5,
                                          keep_isolated_pixels=True)
         assert (result == mask).all()
+
+
+def test_number_of_islands():
```

Comment:
You should also test for clusters that have a radius > 2, to check if the neighbor adding works.